File: mybatis/src/main/java/data/spring/mybatis/application/required/ProductRepository.java
package data.spring.mybatis.application.required;

import data.spring.mybatis.adapter.out.persistence.ProductEntity;
import data.spring.mybatis.application.service.command.ProductSearchCommand;
import data.spring.mybatis.application.service.command.ProductUpdateCommand;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    void save(ProductEntity productEntity);

    int saveAll(List<ProductEntity> productEntities);

    void update(ProductUpdateCommand updateCommand);

    Optional<ProductEntity> findById(Long productId);

    List<ProductEntity> findAll(ProductSearchCommand searchCommand);

    int deleteAll();
}

File: src/main/kotlin/data/spring/mybatis/application/required/product/ProductRepository.kt
package data.spring.mybatis.application.required.product

import data.spring.mybatis.application.service.product.command.ProductSearchCond
import data.spring.mybatis.domain.product.Product
import java.time.LocalDateTime

interface ProductRepository {
    fun save(product: Product): Int

    fun truncate(): Int

    fun findById(productId: Long): Product?
    fun findByCond(searchCond: ProductSearchCond?,
                   createdAt: LocalDateTime?,
                   productId: Long?,
                   size: Int): List<Product>
    fun findAll(): List<Product>
}